enhancements/storage/vsphere-configurable-maximum-allowed-number-of-block-volumes-per-node.md

```diff
@@ -0,0 +1,229 @@
+     variable with the configured limit value into the driver node pods on worker nodes.
+4. Driver reflects the limit in deployments:
+   - The vSphere CSI driver checks that `max-pvscsi-targets-per-vm` FSS is set to true and reflects the limit as
+     `MAX_VOLUMES_PER_NODE` environment variable and uses the configured limit during
+     volume provisioning requests.
+
+### API Extensions
+
+- New field in ClusterCSIDriver CRD:
+  - A new CRD field will be introduced to represent the maximum volume limit configuration.
+  - This CRD will contain a single new field (e.g., `spec.driverConfig.vSphere.maxAllowedBlockVolumesPerNode`) to define
+    the desired limit.
+  - The field will default to the current maximum limit of 59 volumes per node for vSphere 7.
+  - The CRD will validate the value fits within the defined range (1-255).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+No unique considerations for Hypershift. The configuration and behavior of the vSphere CSI driver with respect to the
+maximum volume limit will remain consistent across standalone and managed clusters.
+
+#### Standalone Clusters
+
+This enhancement is fully applicable to standalone OpenShift clusters.
+
+#### Single-node Deployments or MicroShift
+
+No unique considerations for MicroShift. The configuration and behavior of the vSphere CSI driver with respect to the
+maximum volume limit will remain consistent across standalone and SNO/MicroShift clusters.
+
+### Implementation Details/Notes/Constraints
+
+One of the possible future constraints might be increasing the limit with newer vSphere versions. However, we expect the
+limit to be increasing rather than decreasing and making the API validation more relaxed is possible.
+
+### Risks and Mitigations
+
+- None.
+
+### Drawbacks
+
+- Increased Complexity: Introducing a new CRD and operator logic adds complexity to the vSphere CSI driver ecosystem.
+- Potential for Configuration Errors: Incorrectly configuring the volume limit can lead to unexpected behavior
+  or pod scheduling failures.
+- Limited Granularity: The current proposal provides a global node-level limit. More fine-grained control
+  (e.g., per-namespace or per-workload limits) would require further investigation and development.
+
+## Open Questions [optional]
+
+None.
+
+## Test Plan
+
+- E2E tests will be implemented to verify the correct propagation of the configured limit to the driver pods.
+  These tests will be executed only on vSphere 8.
+
+## Graduation Criteria
+
+- GA in 4.19.
+- E2E tests are implemented and passing. 
+- Documentation is updated.
+
+### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+
+### Tech Preview -> GA
+
+- E2E test coverage demonstrating stability.
+- Available by default.
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/).
```

Comment:
Should we mention that we want to wait for VMware to document the configuration on vCenter side?

Reply:
Since this is a criteria for GA I think we should make that clear here - added.